plugin/wavesurfer.playlistparser.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/* Playlist Plugin */
+WaveSurfer.Playlist = {
+    init: function (params) {
+        this.params = params;
+
+        var wavesurfer = this.wavesurfer = params.wavesurfer;
+
+        if (!this.wavesurfer) {
+            throw new Error('No WaveSurfer instance provided');
+        }
+
+        // parse playlist and set params
+        this.playlistFileGET = this.params.playlistFile || null;
+        this.playlistType = this.params.playlistType || 'm3u';
+        this.playlistData = [];
+
+        if (this.playlistFileGET != null) {
+            var ajaxData = wavesurfer.util.ajax({
+                url: this.playlistFileGET,
+                responseType: 'text'
+            });
+
+            var _this = this;
+            ajaxData.on('success', function (data, e) {
+                _this.parse(data);
+                _this.wavesurfer.fireEvent('playlist-ready');
+            });
+            ajaxData.on('error', function (e) {
+                throw new Error('Error reading the playlist file' + 'XHR error: ' + e.target.statusText);
+            });
+
+        } else {
+            throw new Error('No playlist file provided');
+        }
+    },
+
+    parsedList: function() {
+        return this.playlistData;
+    },
+
+    parse: function(playlistFile) {
```

Comment:
parse sounds too techy, I'd personally call it load. And I also prefer `playlistData` over `parsedList`, can you come up with a better name that avoids `parse`?